src/nbl/ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -968,31 +968,24 @@ void CMitsubaLoader::cacheEmissionProfile(SContext& ctx, const CElementEmissionP
 {
 	if (!profile)
 		return;
-
-	const auto cacheKey = ctx.emissionProfileCacheKey(*profile);
-
+	
 	const asset::IAsset::E_TYPE types[]{ asset::IAsset::ET_IMAGE_VIEW, asset::IAsset::ET_TERMINATING_ZERO };
-
-	if (ctx.override_->findCachedAsset(cacheKey, types, ctx.inner, 0u).getContents().empty())
+	auto assetLoaded = ctx.override_->findCachedAsset(profile->filename, types, ctx.inner, 0u);
+	if (assetLoaded.getContents().empty())
 	{
 		auto assetLoaded = interm_getAssetInHierarchy(m_assetMgr, profile->filename, ctx.inner.params, 0u, ctx.override_);
-
-	
-		auto contentRange = assetLoaded.getContents();
-		if (contentRange.empty())
+		if (assetLoaded.getContents().empty())
```

Comment:
I think you still have duplicate work here: https://github.com/Devsh-Graphics-Programming/Nabla/pull/630#discussion_r1457951527

Reply:
[resolved](https://github.com/Devsh-Graphics-Programming/Nabla/commit/b1ac324212a4b8dc3bf98a4797c58266f1ffc19a)